src/main/scala/smack/messaging/Client.scala

```diff
@@ -27,14 +29,19 @@ object Client {
   case class Domain(value: String) extends AnyVal
   case class UserWithoutDomain(value: String) extends AnyVal
   case class UserWithDomain(value: String) extends AnyVal
+  case class MessageId(value: String) extends AnyVal
 
   sealed trait State
   case object Unconnected extends State
   case object Connected extends State
 
+  case class ChatState(
+    channel: Chat,
+    unackMessages: Set[MessageId])
```

Comment:
as this is a sample implementation, should we just store all messages in the ChatState? i.e. replace `unackMessages` with `messages: Seq[MessageState]` where MessageState has the message itself and a status like 'acked, unacked' etc?